### ISSUE TITLE
Display of sms text

### ISSUE THREAD

**Issue by mattridrm** [4IceG/luci-app-sms-tool#3]
I am using openwrt snapshot. No text are displayed and error on page is Uncaught ReferenceError: refreshSMS is not defined
    onclick http://192.168.1.1/cgi-bin/luci/admin/modem/sms:1. Messages are displayed in array on page when inspecting page.

**Comment by 4IceG** [comment]
Hi, what version are you using exactly, the bug is on the last snapshot? Did you install packages yourself or compiled? Do you have any error screen? strange because I still display messages ok, nothing with the error. Have you also tried different ports for communication?

**Comment by mattridrm** [comment]
Compiled my self from Saturdays Openwrt snapshot. I don't think it's the
port as I can see that it lists the text messages in a javascript array
when I view the source of the page in Firefox. I'm not to familiar with
luci to debug but it seems like a javascript error? Any way thanks for your
work. I'm using the atinout app and that is working well.

**Comment by 4IceG** [comment]
The bug looks java but to make it more fun I am not able to recognize it. I did the update and I just compiled the image and packages, everything works. Try to compile again, download new sources, maybe check if it's the same after installing the .ipk package. I attach my sources and ipk packages.

Good luck, it has to work.
[Error.tar.gz](https://github.com/4IceG/luci-app-sms-tool/files/5724962/Error.tar.gz)

**Comment by mattridrm** [comment]
Thanks Rafal. I tried your attached ipkg and still had problems. Did some
more troubleshooting and found the smtool output of the json wasn't
escaping a string in one of the texts from the sim card. Deleted all texts
and app is no working properly. Im not really sure what was causing the
text message to not be escaped properly. Thanks for your work.

**Comment by 4IceG** [comment]
Hi mattridrm if the sms-tool doesn't work you can always try smstools3
and gui for it.

https://github.com/koshev-msk/luci-app-smstools3
or
https://eko.one.pl/forum/viewtopic.php?id=15382&p=2

2020-12-22 0:14 GMT+01:00, mattridrm <notifications@github.com>:

**Comment by mattridrm** [comment]
Thanks Rafal. I did get it to work. I had to delete all of the existing
texts with sms-tool from the modem. Don't know what was hanging up in the
modem and log to cause it. The funny thing is I installed it on another
board with a different Sierra Wireless modem and the luci-app-sms-tool had
the same problem. Wouldn't display texts and the webpage would take forever
to load. A quick sms-tool delete all messages and the app worked properly.
Thanks again for the app.
